### DISCUSSION TITLE
My input values won't update

### DISCUSSION THREAD

**Discussion by MagnusDot** [livewire/livewire#7601] [1. Help] [answered]
since always on my project with simply laravel livewire and alpine, when I make an update of value in an input, nothing wants to be updated, in the debug bar, I see well the change to be made, but nothing in the input itself, the value remains well as with the old value 

```PHP

<?php

namespace App\Livewire\Components;

use Livewire\Component;

class SearchClass extends Component
{
    public $search;


    public function change($test){
        dd($test);
    }

    public function updatedSearch(){
        $this->search = 'oui oui';
    }

    public function render()
    {
        return view('livewire.components.search-class');
    }
}

```

```html

<div class="flex flex-col w-full h-full items-center justify-center text-gray-300 px-16">
    <input wire:model.live="search">
    <button wire:click="change('hello')">test 1</button>
    <button wire:click="change('bye bye')">test 2</button>
</div>

```

the video of nothing ....

can someone help me pls :'( 

https://github.com/livewire/livewire/assets/39904918/bea956c3-b39e-4a1f-9512-c7219c628e44

**Comment by dudelisius** [comment]
Are you expecting that the value for the input changes to 'oui oui' when typing into the input field? 

Have you tried `updatingSearch` instead of `updatedSearch`?

What is it what you are trying to achieve?

**Reply by MagnusDot** [reply]
@dudelisius  I've been able to try updatingSearch or updatedSearch and that's not working, the problem is updating the front-end input. I'm trying to do a select with custom search and result in a dropdown and I need it for my selection to be displayed correctly.

**Reply by dudelisius** [reply]
If I'm understanding correctly you could remove the `updatedSearch` function and display the value with {{ $search }}. This should reflect the users input.

Also If you are trying to build a select component I would recommend looking at some alpine components; 

https://alpinejs.dev/component/select2
https://alpinejs.dev/component/choices

**Reply by MagnusDot** [reply]
it's a really ugly thing to do, given that livewire normally allows an update to be correctly reflected in an input

**Reply by dudelisius** [reply]
I didn't mean to display it in the input, that is definitely wrong.

The search model currently does reflect the input value correctly.

I meant using it somewhere else in the code. You should remove the updated search function.

What would you like to do with the search variable precisly?

**Comment by MagnusDot** [comment]
@dudelisius  the example in the code isn't at all what I want to do, I just want it to put the text I want on my input when I click on a button, but at the moment it doesn't work, livewire doesn't seem to want to do it

**Reply by dudelisius** [reply]
Ah, I understand. Then you should only need the change function and set the value from there like this;

`public function change($test){
    $this->search = $test;
 } `

I would advice to type hint the variable and set the default to an empty string like this;

`public string $search = '';`


Another approach would be to remove the change function as well and set the `search` variable with `$set` like this;

`<button type="button" wire:click="$set('search', 'oui oui');">set search</button>`


Currently on mobile so couldn't test it, I'll see if I can get around to it one of these days, but it is a lot more clear what you want.

**Reply by dudelisius** [reply]
@MagnusDot I think this is what you want right? 

https://github.com/livewire/livewire/assets/394451/34059f0a-2443-4759-a3b9-6707d8de8903

The component code for this is; 

```
<?php

declare(strict_types=1);

namespace App\Livewire;

use Livewire\Component;

class Example extends Component
{	
	public string $search = '';

	public function render()
	{
		return view('livewire.example');
	}
}
```

And the blade file looks like this;

```
<div>
	<div class="w-72 mx-auto bg-white shadow rounded-lg p-10">

		<input wire:model.live="search">

		<button type="button" wire:click="$set('search', 'Hello')" class="rounded-lg border border-gray-500 mt-4 px-3 py-2 hover:bg-gray-100">Change text to "Hello"</button>

		<button type="button" wire:click="$set('search', 'Oui Oui')" class="rounded-lg border border-gray-500 mt-4 px-3 py-2 hover:bg-gray-100">Change text to "Oui Oui"</button>

	</div>
</div>
```

**Reply by MagnusDot** [reply]
https://github.com/livewire/livewire/assets/39904918/7d6fd208-18bb-4c46-9aa8-7b5837e320c2

that's the problem, you're code is not working on my setup, I don't know why it's only happening on my project, I can't work out where it's coming from, it's not working on my end

@dudelisius

**Reply by joshhanley** [reply, marked as answer]
@MagnusDot do you have AlpineJS in your `app.js` file? If so, you need to remove it and use Livewire's inbuilt version of Alpine.

**Reply by dudelisius** [reply]
It looks like Livewire isn't properly installed. Do you see any console.log messages? And can you see the request being made back to the backend in the network tab?

Which versions are you using for Laravel and Livewire??

**Reply by MagnusDot** [reply]
damn! that was it, I feel really stupid, I had added alpine as well as the old versions of livewire ... I'm sorry for the time I've wasted and thanks a lot for your help guys!

**Reply by joshhanley** [reply]
@MagnusDot no worries! 🙂 Just to confirm, did you upgrade an existing application to V3 or is it a new install that you added Alpine to? The reason I ask, is a few people are getting tripped up by this, but we're not clear on how people end up with Alpine loaded separately, so any information is helpful.

**Reply by MagnusDot** [reply]
@joshhanley it's a new app, directly from V3, the thing that's really confusing is that before we had the view in the app.js of the alpine load and now it's done magically

**Reply by joshhanley** [reply]
@MagnusDot ok thanks, good to know. Did you use Laravel Breeze or anything like that? No sure what you mean about the bit that's confusing, can you explain that a bit further?

**Comment by MagnusDot** [comment]
thx for all everyone !
